components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,179 @@
+      cookbook_versions: [{
+        id: 1,
+        name: 'aix',
+        version: '2.3.4',
+        operator: '<'
+      }],
+      default_attributes: {
+        test: 'test'
+      },
+      override_attributes: {
+        test: 'test'
+      }
+    };
+
+    beforeEach(() => {
+      store = TestBed.inject(Store);
+    });
+
+    it('opening create modal and reset tab to details tab', () => {
+      component.visible = true;
+      expect(component.detailsTab).toBe(true);
+      expect(component.constraintsTab).toBe(false);
+      expect(component.defaultTab).toBe(false);
+      expect(component.overrideTab).toBe(false);
+    });
+
+    it('opening create modal resets name, description, default, override to empty string', () => {
+      component.visible = true;
+      expect(component.detailsFormGroup.controls['name'].value).toEqual('');
+      expect(component.detailsFormGroup.controls['description'].value).toEqual('');
+      expect(component.defaultAttrFormGroup.controls['default'].value).toEqual('');
+      expect(component.overrideAttrFormGroup.controls['override'].value).toEqual('');
+    });
+
+    it('on conflict error, modal remains open and displays conflict error', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.detailsFormGroup.controls['name'].setValue(environment.name);
+      component.detailsFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.defaultAttrFormGroup.controls['default'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.overrideAttrFormGroup.controls['override'].setValue(
+        JSON.stringify(environment.override_attributes));
+
+      component.createEnvironment();
+
+      const conflict = <HttpErrorResponse>{
+        status: HttpStatus.CONFLICT,
+        ok: false
+      };
+      store.dispatch(new CreateEnvironmentFailure(conflict));
+      expect(component.conflictError).toBe(true);
+    });
+
+    it('on success, closes modal and adds new environment', () => {
+      spyOn(component.conflictErrorEvent, 'emit');
+      component.visible = true;
+      component.detailsFormGroup.controls['name'].setValue(environment.name);
+      component.detailsFormGroup.controls['description'].setValue(environment.description);
+      component.constraintArray = environment.cookbook_versions;
+      component.defaultAttrFormGroup.controls['default'].setValue(
+        JSON.stringify(environment.default_attributes));
+      component.overrideAttrFormGroup.controls['override'].setValue(
+        JSON.stringify(environment.override_attributes));
+      component.createEnvironment();
+
+      store.dispatch(new CreateEnvironmentSuccess({environment}));
+      expect(component.creating).toBe(false);
+      expect(component.visible).toBe(false);
+      expect(component.environmentsList.length).not.toBeNull();
```

Comment:
This is not confirming what the test purports: that the specific new environment has been added.

Reply:
Agreed. :)